File: frontend/src/app/value-objects/SignUpForm.ts
export class SignUpForm {

  private firstName: string = "";

  private lastName: string = "";

  private email: string = "";

  private password: string = ""

  public static EMAIL_REGEXP = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

  public constructor(init?: Partial<SignUpForm>) {
    Object.assign(this, init);
  }

  isValid() {
    return this.password?.length > 3 && this.firstName !== "" && this.lastName !== "" && SignUpForm.EMAIL_REGEXP.test(this.email)
  }
}

File: backend/src/main/java/app/vo/SignUpForm.java
package app.vo;

import lombok.Data;

import javax.validation.constraints.Email;
import javax.validation.constraints.Size;

@Data
public class SignUpForm {
    @Size(min = 3, max = 100)
    private String password;
    @Email(message = "Email should be valid")
    private String email;
    @Size(max = 50)
    private String firstName;
    @Size(max = 50)
    private String lastName;
}
